cpp/oneapi/dal/backend/linalg/io.hpp

```diff
@@ -0,0 +1,101 @@
+/*******************************************************************************
+* Copyright 2020-2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include <iomanip>
+#include <ostream>
+
+#include "oneapi/dal/backend/linalg/matrix.hpp"
+
+namespace oneapi::dal::backend::linalg {
+
+template <typename T, layout lyt>
+class matrix_formatter {
```

Comment:
Is it used for tests only? If yes, it is better to move it to the test-system-related folder, I guess

Reply:
All linalg stuff is in `test/engine` now